exporter.go

```diff
@@ -151,11 +167,75 @@ func (e *Exporter) Export(
 	return saveImage(workingImage, additionalNames, e.Logger)
 }
 
-func (e *Exporter) addLayer(image imgutil.Image, layer identifiableLayer, previousSHA string) (string, error) {
+func (e *Exporter) Cache(layersDir string, cacheStore Cache) error {
+	var err error
+	origMeta, err := cacheStore.RetrieveMetadata()
+	if err != nil {
+		return errors.Wrap(err, "metadata for previous cache")
+	}
+	meta := cache.Metadata{}
+
+	for _, bp := range e.Buildpacks {
+		bpDir, err := readBuildpackLayersDir(layersDir, bp)
+		if err != nil {
+			return errors.Wrapf(err, "reading layers for buildpack '%s'", bp.ID)
+		}
+
+		bpMD := metadata.BuildpackLayersMetadata{
+			ID:      bp.ID,
+			Version: bp.Version,
+			Layers:  map[string]metadata.BuildpackLayerMetadata{},
+		}
+		for _, layer := range bpDir.findLayers(cached) {
+			layer := layer
```

Comment:
Is this just future-proofing against additional closures that may be introduced and executed in parallel?

Reply:
One of the checks that ran during make reported a potential(?) error when using '&layer' below when the variable wasn't explicitly reassigned. But parallelizing the layer export something I want to explore as well.